File: VDFEditor/src/main/java/de/treichels/hott/vdfeditor/ui/TransmitterTask.java
package de.treichels.hott.vdfeditor.ui;

import de.treichels.hott.HoTTSerialPort;
import gde.model.serial.UpdateHandler;
import gde.model.voice.VoiceFile;
import javafx.concurrent.Task;

public abstract class TransmitterTask extends Task<VoiceFile> implements UpdateHandler {
    private HoTTSerialPort port;

    TransmitterTask(final String title) {
        updateTitle(title);
    }

    public HoTTSerialPort getPort() {
        return port;
    }

    public void setPort(final HoTTSerialPort port) {
        this.port = port;
    }

    @Override
    public void update(final int step, final int count) {
        updateMessage(String.format("%d %%", step * 100 / count));
        updateProgress(step, count);
    }
}

File: VDFEditor/src/main/java/de/treichels/hott/vdfeditor/ui/transmitter/TransmitterTask.kt
package de.treichels.hott.vdfeditor.ui.transmitter

import de.treichels.hott.HoTTSerialPort
import gde.model.serial.UpdateHandler
import gde.model.voice.VoiceFile
import javafx.concurrent.Task

abstract class TransmitterTask internal constructor(title: String) : Task<VoiceFile?>(), UpdateHandler {
    var port: HoTTSerialPort? = null

    init {
        this.updateTitle(title)
    }

    override fun update(step: Int, count: Int) {
        updateMessage(String.format("%d %%", step * 100 / count))
        updateProgress(step.toLong(), count.toLong())
    }
}
